src/drivers/imu/mpu6000/mpu6000.cpp

```diff
@@ -2020,11 +2020,16 @@ MPU6000::measure()
 	arb.scaling = _accel_range_scale;
 	arb.range_m_s2 = _accel_range_m_s2;
 
+	// We calculate sensor temperature as :
+	// temperature_real = temperature_raw * temperature_sensitivity + zero_offset
+	//
+	// Sensitivity (in degC/LSB) is from the Invensense datasheet for the specific sensor
+	// Zero offset (in degC) varies from part to part, and these values are good defaults
 	if (is_icm_device()) { // if it is an ICM20608
 		_last_temperature = (report.temp) / 326.8f + 25.0f;
 
 	} else { // If it is an MPU6000
-		_last_temperature = (report.temp) / 361.0f + 35.0f;
+		_last_temperature = (report.temp) / 340.0f + 36.0f;
```

Comment:
Is the temp offset not 35 degC for the MPU6000?

![mpu6000_temp_sensitivity](https://user-images.githubusercontent.com/4036885/37334972-a7189bc8-26ad-11e8-8fa9-da84237b722a.png)

Reply:
Thanks, you are indeed right. I believe I had been looking at an older version of the datasheet. Will fix.

Reply:
Do you know whether this sensitivity is related to a particular MPU6000 silicon revision (as in we will have to handle the temp conversion based on the silicon revision)? Or is it that there was a typo in the previous data sheets and they corrected it in the later version. 

I am also not sure if I have the latest datasheets. These were the ones that I downloaded from the Invensense website, so as long as those are the most recent, then it should be alright.